vm-linux-helpers.cmake

```diff
@@ -52,6 +52,46 @@ function(AddFileToOverlayDir filename file_location root_location overlay_name)
     )
 endfunction(AddFileToOverlayDir)
 
+# Function to add a symbolic link to an overlay directory.
+# filename: The name of the link in the overlay
```

Comment:
Actually, why do we have `root_location` and `file_location`as separate parameters? We could merge this into one parameter and extract the path internally. Then there is less that could be used wrongly

```    
# symlink: The link to create in the overlay (path/to/link)
# file: The file where the link points to (path/to/file)

...

get_filename_component(symlink symlink_dir DIRECTORY)
get_filename_component(symlink symlink_name NAME)
set(symlink_dir "${rootfs_output_dir}/${symlink_dir}")
add_custom_command(
        OUTPUT ${rootfs_output_dir}/${root_location}/${filename}
        COMMAND
            cd ${rootfs_output_dir} \
            && if [ ! -d ${symlink_dir} ]; then mkdir -p ${symlink_dir}; fi \
            && cd ${symlink_dir} &&  ln -s ${file} ${symlink_name}
        VERBATIM
        DEPENDS ${ROOTFS_FILE_OVERLAY_DEPENDS}
```

Reply:
This is just following the pattern established by `AddFileToOverlayDir` above.